### ISSUE TITLE
How to dynamically load children nodes in TreeComponent?

### ISSUE THREAD

**Issue by alexandis** [abpframework/abp#12867]
Hi.

You made use of Nz Tree component, but as I can see you have hidden a lot of features of this component. You have not supplied your TreeComponent with documentation either.
What I need now is:
1) how to prevent node deselecting?
2) how to dynamically load children nodes for the selected node? By default, I output only a root level of nodes. But in this case there is no ability to expand such nodes. So I cannot load its children dynamically... Probably this property should have been passed through:

![image](https://user-images.githubusercontent.com/6149843/171695412-25e0c908-84b0-4ed0-bad4-5deb1c55c26e.png)

**Comment by alexandis** [comment]
I've already implemented #2 using custom button handling custom handler:

```
          const childComponentNodes = componentsResponse?.items.map(item => ({
            id: item.componentId,
            parentId: item.parentComponentId,
            displayName: `${item.componentId} | ${item.description} | ${item.componentType.description}`,
            component: item
          } as Components.ComponentNode));
          this.treeAdapter.handleUpdate({
            key: node.id,
            children: childComponentNodes
          });
          this.nodes = this.treeAdapter.getTree();
```

Question #1  remains.

**Comment by alexandis** [comment]
Small asking: I would prefer to be able to modify `abpTreeExpandedIconTemplate` in a way that I can use any content there - for example, to show 'plus' sign to load child nodes (meanwhile it displays any content ONLY if there ARE already child nodes loaded). Instead of adding custom button to the `abpTreeNodeTemplate` to make it possible. The latter approach breaks the layout and user experience...
